platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/PathToFileResolverCodec.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.serialize.codecs.core
+
+import org.gradle.api.internal.file.FileLookup
+import org.gradle.internal.file.PathToFileResolver
+import org.gradle.internal.serialize.graph.Codec
+import org.gradle.internal.serialize.graph.ReadContext
+import org.gradle.internal.serialize.graph.WriteContext
+import org.gradle.internal.serialize.graph.readFile
+import org.gradle.internal.serialize.graph.writeFile
+
+
+class PathToFileResolverCodec(
+    private val fileLookup: FileLookup,
+) : Codec<PathToFileResolver> {
+
+    override suspend fun WriteContext.encode(value: PathToFileResolver) {
+        val baseDir = value.baseDir
```

Comment:
💭 Rather than augmenting the `PathToFileResolver` to expose the `baseDir`, have you considered type casing the concrete implementations like we for file collections, for example?

Reply:
I responded in another comment mostly. The simplistic approach didn't pan out, so I'll update the codec to rely on concrete implementations